### DISCUSSION TITLE
Build: Skip steps

### DISCUSSION THREAD

**Discussion by zalito12** [withastro/roadmap#1024] [Proposal]
### Body

# Summary

Add flags to cli to allow skip steps on build process.

# Background & Motivation

We all know that the build process is the bottlekneck of the Astro awesomeness, the incremental build to allow build just one single page/route/document will be the icing on the cake, but what about when you change just one small detail of your project, why do every build  step.

I was building my first project and I changed  a small detail on the template of a long collection, but the images and their sizes didn't change, I would like to skip the image optimization step.

# Goals

- Make re-builds faster
- Skip image optimization on build

# Example

Add some kind of flag  to skip image optimization like "--skip-images".
I don't know how build works, but I guess it would involve some kind of cache to keep track of the already existing images hashes on the html generation.

**Comment by Princesseuh** [comment]
Images are already cached between builds

**Reply by zalito12** [reply]
Wow, That's awesome, as I said I don't too much about the background  working of the build process, my apologies.
